src/Native/Runtime/arm64/Interlocked.S

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+//
+
+#include <unixasmmacros.inc>
+
+// WARNING: Code in EHHelpers.cpp makes assumptions about this helper, in particular:
+// - Function "InWriteBarrierHelper" assumes an AV due to passed in null pointer will happen at RhpLockCmpXchg32AVLocation
+// - Function "UnwindWriteBarrierToCaller" assumes the stack contains just the pushed return address
+// x0 = destination address
+// w1 = value
+// w2 = comparand
+LEAF_ENTRY RhpLockCmpXchg32, _TEXT
+    mov     x8, x0          // Save value of x0 into x8 as x0 is used for the return value
+ALTERNATE_ENTRY RhpLockCmpXchg32AVLocation
```

Comment:
All places in C++/asm runtime code where null reference exceptions are expected have to be whitelisted with `AVLocation` label like this.

It makes it possible to differentiate between expected segfaults that should be converted into managed NullReferenceException, and fatal segfaults (e.g. segfault during GC) that should result into immediate abort / core dump.